### ISSUE TITLE
Inconsistent GUI scaling on HiDPI displays

### ISSUE THREAD

**Issue by z80maniac** [BiglySoftware/BiglyBT#479]
I think that this is different from https://github.com/BiglySoftware/BiglyBT/issues/16. That issue is about icons, and this issue is about everything else.

![screenshot_20180526_212110](https://user-images.githubusercontent.com/20016485/40579531-c6ad6d4c-6131-11e8-805e-6de51a4071f3.png)

Monitor: 3840×2160 pixels, 23.5 inches
OS: Kubuntu 18.04
BiglyBT: v1.5.0.0

There are several problems:
1. Icons for the upper buttons are not in place.
2. Search placeholder is tiny. And when you start typing in that field, the entered text is also the same size.
3. Search control is disproportionately tall.
4. Labels on the left pane are tiny.
5. Not on the screenshot, but the text in Dashboard's browser is barely readable because it's very small too.

Except for the above problems, almost everything else is scaled correctly (Options, Plugins & Extras, Devices).

None of these problems can be solved by changing DPI setting in Options. That setting only partially changes the scaling for the main pane and some tables, but the rest of GUI stays mostly the same or just stretches in only one dimension. See the example for explicit 360dpi setting:

![screenshot_20180526_213800](https://user-images.githubusercontent.com/20016485/40579536-de96525c-6131-11e8-9796-2b6a71bb3dfc.png)

You can see that the scaling was inconsistent. Some icons have been scaled up, some - not. Some text has been scaled up, some - not. The left pane was scaled horizontally (yes, that's the minimum allowed width for it at 360dpi), but the labels remained tiny.

**Comment by MagnumGR** [comment]
I have a similar problem with BiglyBT (also with Azureus).
The problem appears after upgrades to latest Fedora 29. 

![bigly](https://user-images.githubusercontent.com/8311546/49695152-1b3aa480-fb9f-11e8-8044-ee46c5ae8ed5.png)

**Comment by TuxPaper** [comment]
Can you paste the info in the "System" of the About Window? 

We made improvements in 1.7.0.0.  If you are already on 1.7, try joining the beta (Help->Join Beta Porgram) and see if that changes anything

**Comment by MagnumGR** [comment]
![biglybt-info](https://user-images.githubusercontent.com/8311546/49720463-7d131100-fc68-11e8-97e4-215b3191f9bf.png)
Later i will try Beta version of BigLy

**Comment by MagnumGR** [comment]
![biglybtbeta](https://user-images.githubusercontent.com/8311546/49734523-3daaeb80-fc8d-11e8-8ccd-3ca555830905.png)

This is from Beta 1.7.0.1-B24.  

The symbols like (>) and height of the bar is back to normal, but now the fonts of the bars, Categories & Search Box is very very small

**Comment by TuxPaper** [comment]
The beta has more information in the about window (zoom and dpi)

I tried with a similar FC29 version, and here's how it displays for me:

![image](https://user-images.githubusercontent.com/725353/49839201-e70fe100-fd62-11e8-9a6f-c511980ab65d.png)

"Search library" is definitely too small, but the tag buttons are ok on mine.  Do you know if you changed any display settings (theme, fonts, etc)?

Also, if you can report the new zoom and dpi fields from the about window, maybe they are different than mine.

**Comment by MagnumGR** [comment]
![biglybt-beta](https://user-images.githubusercontent.com/8311546/49855800-e45ecb80-fdf6-11e8-88ab-6fc37a7149f9.png)

This is mine... The only different is dpi, i don't know if this matter but i have a radeon GPU with the following settings
![display](https://user-images.githubusercontent.com/8311546/49855865-1a9c4b00-fdf7-11e8-9a5e-237f0812ebe9.png)

**Comment by z80maniac** [comment]
FWIW, here are my screenshots:
![screenshot_20181212_114620](https://user-images.githubusercontent.com/20016485/49858429-ef6d2800-fe05-11e8-91f3-62cfe57a9739.png)
![screenshot_20181212_115041](https://user-images.githubusercontent.com/20016485/49860582-272a9e80-fe0b-11e8-9477-ddf8b30ec3f1.png)
![image](https://user-images.githubusercontent.com/20016485/49859927-c8b0f080-fe09-11e8-9255-576ec58a4890.png)
![image](https://user-images.githubusercontent.com/20016485/49860102-28a79700-fe0a-11e8-9170-c3f8a29f7b86.png)
Basically, almost everything is relatively ok now, except the search field is a bit small and the buttons below the search bar are almost invisible.

Also the installer is barely readable:
![image](https://user-images.githubusercontent.com/20016485/49859696-3872ab80-fe09-11e8-8977-7316216ad80a.png)

**Comment by TuxPaper** [comment]
Can you try the B28 and see how things are?  I've changed a lot of the fonts to relative sizes, and fixed up some others to ensure they should* fit the bounds they are constrained to.

**Comment by MagnumGR** [comment]
![biglybt-b28](https://user-images.githubusercontent.com/8311546/50051335-a1fbfe00-0118-11e9-9f88-36db5fc38140.png)

Looks with B28 everything is back to normal.
Thanks !

**Comment by z80maniac** [comment]
Now the scaling in this window is ok:
![image](https://user-images.githubusercontent.com/20016485/50085928-97229580-020c-11e9-9061-a14bfe33ea28.png)
The "All" button is white on a white background, but I guess it has something to do with theme support and not DPI, so it's out of scope.
The browser text is still very small though:
![image](https://user-images.githubusercontent.com/20016485/50086172-45c6d600-020d-11e9-865f-68e499dbd25e.png)

**Comment by TuxPaper** [comment]
I'll close this issue, even though there is bound to be more scaling issues (like the installer).  The installer is 3rd Party, so we are limited in what we can fix (the installer does have a non-GUI install mode), but other issues can be filed as they are found.

Thanks for all your help and screenshots. A++, would fix again.
